GitUI/CommandsDialogs/FormPull.cs

```diff
@@ -557,6 +557,26 @@ void PopStash()
                     UICommands.StashPop(owner);
                 }
             }
+
+            void executeBeforeScripts()
+            {
+                if (!Fetch.Checked)
```

Comment:
`git pull` is a shortcut for `git fetch + git merge`, so "fetch" is always run, and the "merge" part is only run for "pull" (in this case `!Fetch`).

Reply:
As RussKie mentioned, the fetch step ist part of the pull and therefore "locked in" by the pull Before/After. 
You are right that the change is kind of "incompatible" since the old behaviour was to execute Before/AfterPull also on pure fetches (which IMO is not correct).

What I found kind of confusing is that the Before/AfterMerge scripts are not executed on pull (which is fetch+merge). Shouldn't a complete call-chain for pull be:
BeforePull - BeforeFetch - _fetch_ - AfterFetch - BeforeMerge - _merge_ - AfterMerge - AfterPull
?
Since git pull is a predefined command it would be a bit hard (and not wanted) to disassemble this, so it might be sufficient to consider the pull-merge as an own operation handled by the AfterPull.

Reply:
> git pull is a shortcut for git fetch + git merge, 

Sure, but the script runs in relation to git-fetch and git-git-pull. The call chain is not splitting the pull as KarstenRa's example. I have occasionally wanted this, but then is is the fetch-after that I needed (workaround is not so strange, only git-fetch and merge manually). 

I accepted the functionality as is yesterday, but it is easier to explain that the commands are related to the Git commands, and it is normally better to implement something that can be explained.

I really had three observations in one comment here. The second must be resolved first anyway.

Reply:
I added the suggested comment since I get the confusion that `!Fetch.Checked` means fetch+merge in that context.

Reply:
The sequence is BeforePull-BeforeFetch-gitFetch/pull-AfterFetch-AfterPull
Is this what we want?
(I will probably approve anyway, this is not obvious)

Reply:
I think this is the most useful/evident policy as long as the pull is not splitted into fetch and merge (what I think would be a little overkill). I don't see any practical drawbacks in having the AfterFetch executed after the pull-merge (instead of before it), but do not exclude that I am missing a usecase.